content/modules/r/pages/data-frames.adoc

```diff
@@ -0,0 +1,348 @@
+= data.frames
+
+{sp} +
+
+== Abstract
+
+Data.frames are one of the primary data structures used very frequently when working in R. Data.frames are tabular data structures, consisting of named columns of equal size, where each column has a single type.
+
+You can create a data.frame easily. Vectors are used to fill the columns as follows:
+
+{sp} +
+
+=== Bracket Subsetting/Indexing
+
+==== Data Frame Creation
+====
+[source,r]
+----
+df <- data.frame(cat_1=c(1,2,3), cat_2=c(9,8,7), 
+                 ok=c(T, T, F), other=c("first", "second", "third"))
+
+head(df)
+----
+
+----
+   cat_1 cat_2    ok  other
+ 1     1     9  TRUE  first
+ 2     2     8  TRUE second
+ 3     3     7 FALSE  third
+----
+====
+
+The different columns of a data frame can be disparate in type, the only requirement is that values column-wide have the same type. In this case, `cat_1` and `cat_2` contain integers, `ok` contains booleans, and `other` contains Strings.
```

Comment:
The columns in a data.frame can be distinct/different, however, here is one requirement that each column contains the same type of values.